### PR TITLE
add notes for Windows

### PR DESCRIPTION
**Pull request by dirk-thomas**
Addresses the problem described in colcon/colcon-notification#4.

### CODE REVIEW

**Comment by sloretz** [comment]
@dirk-thomas Success!

Mind if I add some commits to this branch? I'd like to add notes about
* installing some dependencies to get the first build to succeed
* adding `COLCON_IGNORE` the argcomplete package since it fails to build on windows
* using curl to download the repos file since wget isn't available

Here's what worked for me

```bat
python -m venv env
call env\Scripts\activate
type nul > env\COLCON_IGNORE
pip install -U EmPy setuptools PyYAML pywin32 coloredlogs

curl --output colcon.repos https://raw.githubusercontent.com/colcon/colcon.readthedocs.org/master/colcon.repos
mkdir src
vcs import src < colcon.repos
type nul > src\colcon-argcomplete\COLCON_IGNORE

python src\colcon-core\bin\colcon build --paths src/*

call "install\setup.bat"
python install\colcon-core\Scripts\colcon-script.py build
```

**Comment by dirk-thomas** [comment]
> Mind if I add some commits to this branch?
> * adding COLCON_IGNORE the argcomplete package since it fails to build on windows
> * using curl to download the repos file since wget isn't available

Sounds good to me.

> * installing some dependencies to get the first build to succeed

I would keep theyat change separate from this. Also I don't think that any of the ones you mentioned are necesssary. They are all mention in the setup files and are being installed already. Anyway since there are many others I would keep it separate to not "block" this one from getting merged soon.

**Comment by dirk-thomas** [comment]
See #11 for a stub.

**Comment by dirk-thomas** [comment]
Since the new "Fetch the sources" instructions for Windows work on all platforms I am fine with keeping them in a single section.

File: developer/bootstrap.rst [hunk 1]
```diff
@@ -23,6 +23,10 @@ While not strictly necessary it is recommended to use a virtual environment for
     $ python3 -m venv /tmp/colcon-venv
     $ . /tmp/colcon-venv/bin/activate
 
+.. note::
+
+    On Windows the executable is likely name ``python`` and the script is located in ``/tmp/colcon-venv/Scripts/activate``.
```
[the remark this answers is not in the document]
Comment (sloretz) [reply developer/bootstrap.rst]: The windows note no longer references `/tmp` in 571e3f3

**Comment by sloretz** [comment]
@mikaelarguedas Common download instructions in f0ecc1c

**Comment by dirk-thomas** [comment]
> We could specify to use call on Windows for sourcing the setup files but I don't think its required to get this merged

I added about the script is being named and called on Windows in a7c192beaff89a391310de25274052a857c6890e (I didn't use `call` for it) and removed it from the activation script in a3229f9e2eae730556f2577018212803cc3a01c6 (assuming that it was not required, maybe @sloretz can confirm?).

**Comment by sloretz** [comment]
@dirk-thomas I just checked and can confirm it works both with and without `call`.

**Comment by dirk-thomas** [comment]
I added another related note in 455b7392408d137a0e91cade6890bba8dc49c255.

**Pull request merged**